bench/darktable/optimized/FileReader.ll

```diff
@@ -146,14 +146,14 @@ define hidden void @_ZNK8rawspeed10FileReader8readFileEv(ptr dead_on_unwind noal
   br label %.body
 
 "_ZNSt10unique_ptrI8_IO_FILEZNK8rawspeed10FileReader8readFileEvE3$_0ED2Ev.exit": ; preds = %28
-  %45 = load ptr, ptr %24, align 8, !tbaa !15, !nonnull !21, !noundef !21
+  %45 = load ptr, ptr %24, align 8, !tbaa !15
```

Comment:
Lost metadata.

Reply:
This is caused when the assume is removed the load have no user and is removed before inline is done, but after inline there is other uses of the pointer.
before this PR as the load was ephemeral the fold was done first after inline when it was not ephemeral any more.
is it possible to only run this fold after inline is done?